api/v1alpha1/rollouthistory_types.go

```diff
@@ -0,0 +1,206 @@
+/*
+Copyright 2022 The Kruise Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
+	networkingv1 "k8s.io/api/networking/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	v1alpha2 "sigs.k8s.io/gateway-api/apis/v1alpha2"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// RolloutHistorySpec defines the desired state of RolloutHistory
+type RolloutHistorySpec struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+
+	// RolloutWrapper indicates information of the rollout related with rollouthistory
+	RolloutWrapper RolloutWrapper `json:"rolloutWrapper,omitempty"`
+
+	// Workload indicates information of the workload, such as cloneset, deployment, advanced statefulset
+	Workload Workload `json:"workload,omitempty"`
+
+	// ServiceWrapper indicates information of the service related with workload
+	ServiceWrapper ServiceWrapper `json:"serviceWrapper,omitempty"`
+
+	// TrafficRoutingWrapper indicates information of traffic route related with workload
+	TrafficRoutingWrapper TrafficRoutingWrapper `json:"trafficRoutingWrapper,omitempty"`
+}
+
+// RolloutWrapper indicates information of the rollout related
+type RolloutWrapper struct {
+	// Name indicates the rollout name
+	Name string `json:"name,omitempty"`
+	// Rollout indecates the related rollout
+	Rollout Rollout `json:"rollout,omitempty"`
```

Comment:
add filed, RolloutID string `json:"rolloutID,omitempty"`, it can be assumed that each rollout-id corresponds to a history CRD.